beacon-chain/rpc/beaconv1/state.go

```diff
@@ -56,3 +100,80 @@ func (bs *Server) GetStateFork(ctx context.Context, req *ethpb.StateRequest) (*e
 func (bs *Server) GetFinalityCheckpoints(ctx context.Context, req *ethpb.StateRequest) (*ethpb.StateFinalityCheckpointResponse, error) {
 	return nil, errors.New("unimplemented")
 }
+
+func (bs *Server) headStateRoot(ctx context.Context) ([]byte, error) {
+	stateRoot, err := bs.ChainInfoFetcher.HeadRoot(ctx)
```

Comment:
`HeadRoot` returns the block root, not state root

Reply:
I think the main issue here is function naming. We have a notion of head block and head state. Having just `HeadRoot` in the name is confusing. I admit that I did not check the implementation, but if we had better naming it wouldn't have been necessary.

Reply:
If you agree with me, I can prepare a PR that makes function names more self-explanatory.

Reply:
In the eth2 terminology land, block is the first citizen, something root (without explicit naming) always refers to the root of the block. Example:

SlotRoot = the root of the block at that slot
EpochRoot = the root of the block at that epoch
CheckpointRoot = the root of the block for that check point

If one do change one then we have to change all for consistency, that's just too verbose imo. Let's leave it as it is.
What we can do is to improve on the comment.

// HeadRoot returns the root of the head of the chain.

Can be better updated to:

// HeadRoot returns the block root of the head of the chain.

Reply:
Thanks for clarifying. I will try to remember this going forward.